src/sage/combinat/lp_algebra_seed.py

```
@@ -0,0 +1,1207 @@
+
+        Nonirreducible polynomials will raise an exception::
+
+            sage: var('x1, x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 4 - x2^2, x2: 1 + x1})
+            Traceback (most recent call last):
+            ...
+            ValueError: (LP2) fail: -x2^2 + 4 is not irreducible over Integer Ring
+
+        Different base rings are allowed::
+
+            sage: var('x1,x2')
+            (x1, x2)
+            sage: S = LPASeed({x1: 1 + x2, x2: 1 + x1}, base_ring=QQ)
+            sage: S
+            A seed with cluster variables [x1, x2] and exchange polynomials [x2 + 1, x1 + 1]
+        """
+        # unpack supplied data
+
+        # make a copy
+        if isinstance(data, LPASeed):
+
+            self._names = data._names
+            self._base_ring = data._base_ring
+            self._coefficients = data._coefficients
+            self._ambient_field = data._ambient_field
+            self._polynomial_ring = data._polynomial_ring
+            self._rank = data._rank
+            self._exchange_polys = copy(data._exchange_polys)
+            self._laurent_polys = copy(data._laurent_polys)
+            self._cluster_vars = copy(data._cluster_vars)
+            self._mutation_sequence = copy(data._mutation_sequence)
+
+        # we assume we have dictionary of variable/poly pairs
+        elif isinstance(data, dict):
+
+            names = tuple(data)  # ensure names are immutable
+
+            # if we are not supplied any coefficients, check input data
+            if not coefficients:
+                for name in names:
+                    # variables present in this polynomial
+                    try:
+                        name_poly_vars = set(SR(data[name]).variables())
+                    except BaseException:
+                        raise TypeError('values of dictionary are not polynomials')
```

```suggestion
                    name_poly_vars = set(SR(data[name]).variables())
```
Just let this fail by itself.

I would *strongly* avoid using the `SR` and instead work with a specific `LaurentPolynomialRing`. It is __much__ faster and more robust.